### PR TITLE
RDM-2790 Fix coverity issues.

### PR DESCRIPTION
**Pull request by andrewfolga**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDM-2790

### Change description ###
Eliminate issues identified by Coverity

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/ccd/PersistenceIT.java [hunk 1]
```diff
@@ -35,12 +35,14 @@ public EmbeddedPostgres embeddedPostgres() throws IOException {
 
         @Bean
         public DataSource dataSource() throws IOException, SQLException {
+            System.out.println("AAAAAAAAAAAAAAA");
```
Comment (dwaynebailey) [inline src/test/java/uk/gov/hmcts/ccd/PersistenceIT.java]: Is this needed?

File: src/main/java/uk/gov/hmcts/ccd/data/definition/DefaultCaseDefinitionRepository.java [hunk 2]
```diff
@@ -129,28 +135,26 @@ public UserRole getUserRoleClassifications(String userRole) {
             final HttpEntity requestEntity = new HttpEntity<CaseType>(securityUtils.authorizationHeaders());
             final Map<String, String> queryParams = new HashMap<>();
             queryParams.put("roles", StringUtils.join(userRoles, ","));
-            return Arrays.asList(restTemplate.exchange(applicationParams.userRolesClassificationsURL(), HttpMethod.GET, requestEntity, UserRole[].class, queryParams).getBody());
+            return Arrays.asList(ofNullable(getClassificationForUserRoleList(requestEntity, queryParams)).orElseThrow(() -> NULL_RESPONSE_EX).getBody());
```
Comment (andrewfolga) [inline src/main/java/uk/gov/hmcts/ccd/data/definition/DefaultCaseDefinitionRepository.java]: Avoid NullPointerException. When no body sent back from server spring will translate the response into null reference (https://stackoverflow.com/questions/21383679/what-happens-when-resttemplate-gets-a-null-response-from-spring).

File: src/main/java/uk/gov/hmcts/ccd/data/definition/DefaultCaseDefinitionRepository.java [hunk 3]
```diff
@@ -129,28 +135,26 @@ public UserRole getUserRoleClassifications(String userRole) {
...
         } catch (Exception e) {
-            LOG.warn("Error while retrieving classification for user roles {} because of ", userRoles, e);
-            throw new ServiceException("Error while retrieving classification for user roles " + userRoles + " because of " + e.getMessage());
+            LOG.warn("Error while retrieving classification for user roles {} because of: ", userRoles, e);
+            throw new ServiceException("Error while retrieving classification for user roles " + userRoles + " because of: " + e.getMessage());
         }
```
Comment (adr1ancho) [inline src/main/java/uk/gov/hmcts/ccd/data/definition/DefaultCaseDefinitionRepository.java]: Looks like you could also use public ServiceException(final String message, final Exception cause)
That way, you do not throw away the stack trace.  What do you think?

File: src/main/java/uk/gov/hmcts/ccd/data/definition/DefaultCaseDefinitionRepository.java [hunk 4]
```diff
@@ -129,28 +135,26 @@ public UserRole getUserRoleClassifications(String userRole) {
...
     }
 
     @Override
     public CaseTypeDefinitionVersion getLatestVersion(String caseTypeId) {
         try {
             final HttpEntity requestEntity = new HttpEntity<CaseType>(securityUtils.authorizationHeaders());
-            CaseTypeDefinitionVersion version = restTemplate.exchange(applicationParams.caseTypeLatestVersionUrl
-                    (caseTypeId), HttpMethod.GET, requestEntity, CaseTypeDefinitionVersion.class).getBody();
+            CaseTypeDefinitionVersion version = ofNullable(getLatestVersion(caseTypeId, requestEntity)).orElseThrow(() -> NULL_RESPONSE_EX).getBody();
```
Comment (adr1ancho) [inline src/main/java/uk/gov/hmcts/ccd/data/definition/DefaultCaseDefinitionRepository.java]: I must admit I find it difficult to browse line longer than 120 characters on my screen

Comment (andrewfolga) [reply src/main/java/uk/gov/hmcts/ccd/data/definition/DefaultCaseDefinitionRepository.java]: I thought we agreed and later changed the line length to 160?

File: src/main/java/uk/gov/hmcts/ccd/data/definition/DefaultCaseDefinitionRepository.java [hunk 5]
```diff
@@ -129,28 +135,26 @@ public UserRole getUserRoleClassifications(String userRole) {
...
             LOG.debug("retrieved latest version for case type: {}: {}", caseTypeId, version);
             return version;
-
         } catch (Exception e) {
             LOG.warn("Error while retrieving case type version", e);
-            if (e instanceof HttpClientErrorException && ((HttpClientErrorException)e).getRawStatusCode() == RESOURCE_NOT_FOUND) {
-                throw new ResourceNotFoundException("Resource not found when getting case type version for " + caseTypeId + " because of " + e.getMessage());
+            if (e instanceof HttpClientErrorException && ((HttpClientErrorException) e).getRawStatusCode() == RESOURCE_NOT_FOUND) {
+                throw new ResourceNotFoundException("Resource not found when getting case type version for " + caseTypeId + " because of: " + e.getMessage());
             } else {
```
Comment (adr1ancho) [inline src/main/java/uk/gov/hmcts/ccd/data/definition/DefaultCaseDefinitionRepository.java]: Just wondering whether we could also refactor ResourceNotFoundException to wrap around Throwable, like 
public RuntimeException(String message, Throwable cause)

**Pull request closed without merging**